beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/handler/UnreferencedHandler.java

```diff
@@ -0,0 +1,46 @@
+package com.expediagroup.beekeeper.cleanup.handler;
+
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.expediagroup.beekeeper.cleanup.path.PathCleaner;
+import com.expediagroup.beekeeper.cleanup.path.aws.S3PathCleaner;
+import com.expediagroup.beekeeper.core.model.EntityHousekeepingPath;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+
+@Component
+public class UnreferencedHandler extends GenericHandler {
+
+  private static final LifecycleEventType EVENT_TYPE = LifecycleEventType.UNREFERENCED;
+  private S3PathCleaner s3PathCleaner;
```

Comment:
Shouldn't be an issue. A question for more context on this though and related to the `HdfsPathCleaner` comment above, do we see the UnreferencedHandler needing to ever cleanup HDFS items? Or is it only ever going to cleanup S3 items?

I was under the assumption that when we introduce the ExpiredHandler it would need the ability to cleanup HDFS & S3 Paths and have them both referenced as separate cleaners, but Unreferenced would not need to have this distinction.

Reply:
Good question. Up to now we've only needed to support S3 but we introduced the interface and S3-specific implementation based on other projects where we've needed to support different file systems. So, I think where possible we should try to be FileSystem-agnostic and allow the FileSystem in use to be chosen by configuration - or at the very least to hard code this for now (e.g. in CommonBeans) but with the idea that it would be fairly easy to introduce another FileSystem later when needed. Hopefully most other classes can just rely on a generic PathCleaner and not need to know whether S3/HDFS/GCS etc. is in use.

Reply:
Reverted to PathCleaner